core/src/main/java/io/grpc/inprocess/InProcessTransport.java

```diff
@@ -664,6 +668,13 @@ public void setMaxInboundMessageSize(int maxSize) {}
 
       @Override
       public void setMaxOutboundMessageSize(int maxSize) {}
+
+      @Override
+      public void setDeadline(Deadline deadline) {
+        headers.discardAll(TIMEOUT_KEY);
+        long effectiveTimeout = max(0, deadline.timeRemaining(TimeUnit.NANOSECONDS));
```

Comment:
This `max()` check can be put in `TIMEOUT_KEY`'s marshaller.

Reply:
Then the marshaller will not look as nice as before because serialization and deserializeation is no longer inverse of each other.

Reply:
Inverse in what way? Like put()ing a `-1` doesn't get() a `-1`, or the method implementations no longer appear symmetric?

Reply:
We spoke offline. It was more the -1 case. Since `TIMEOUT_KEY` has a `< 0` check already, I was okay with leaving this code as-is.